### PR TITLE
Feature/col 488 check expiry

### PR DESCRIPTION
**Pull request by ayang99**
Only 3 files were actually changed as part of this PR. Most changes are from merging PR for COL-525

### CODE REVIEW

**Comment by btiernay** [comment]
Why wouldn't they already be in develop though?

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadService.java [hunk 1]
```diff
@@ -171,6 +172,8 @@ private void resumeIfPossible(DownloadRequest request, boolean checksum)
       spec = downloadStateStore.getProgress(request.getOutputDir(), request.getObjectId());
     } catch (NotRetryableException e) {
       log.info("New download: {} because {}", request.getObjectId(), e.getMessage());
+      terminal.printStatus("Restarting ");
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadService.java]: `Restarting...`?

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadService.java [hunk 2]
```diff
@@ -348,7 +351,7 @@ private String calculateChecksum(File outputFile) {
   }
 
   private String decodeDigest(byte[] digest) {
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadService.java]: Guava:

```java
BaseEncoding.base16().encode(bytes);
``

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadStateStore.java [hunk 3]
```diff
@@ -74,30 +76,47 @@ public boolean hasStarted(File stateDir, String objectId) {
...
   private boolean isCompleted(File stateDir, String objectId, Part part) {
-    File partFile = new File(getObjectStateDir(stateDir, objectId), getPartName(part));
+    val partFile = new File(getObjectStateDir(stateDir, objectId), getPartName(part));
     return partFile.exists();
   }
 
   public void commit(File stateDir, String objectId, Part part) {
     log.debug("Attempting to commit {} part {} to {}", objectId, part.getPartNumber(), stateDir.toString());
-    ObjectMapper mapper = new ObjectMapper();
+    val mapper = new ObjectMapper();
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadStateStore.java]: Relatively expensive to create object mappers. Use `Jackson.DEFAULT` instead, or create static field.

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadStateStore.java [hunk 4]
```diff
@@ -110,9 +129,9 @@ private Part loadPart(File stateDir, String objectId, String partFileName) {
     File objectStateDir = getObjectStateDir(stateDir, objectId);
     File partFile = new File(objectStateDir, partFileName);
 
-    ObjectMapper mapper = new ObjectMapper();
+    val mapper = new ObjectMapper();
     Part part = null;
-    try (FileInputStream partStream = new FileInputStream(partFile)) {
+    try (val partStream = new FileInputStream(partFile)) {
       part = mapper.readValue(partStream, Part.class);
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadStateStore.java]: Can just use `mapper.readValue(partFile, Part.class)`. Would encapsulate this in a helper (i.e. `readPart(partFile))`

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    }
+    return expiry;
+  }
+
+  static boolean isExpired(LocalDateTime expiry) {
+    val now = LocalDateTime.now();
+    log.trace("Now DateTime: %s%n", now);
+    log.trace("Specified Expiry: %s%n", expiry);
+
+    // TOOD: remove this bit
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java]: What were your intentions here?

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    if (expiry.isBefore(now)) {
+      System.out.println("Expired!");
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java]: Should this be going to `stderr` or `Terminal`?

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+    } else {
+      System.out.println("Not expired; still " + ChronoUnit.SECONDS.between(now, expiry) + " seconds to go");
+    }
+    return expiry.isBefore(now);
+  }
+
+  static Map<String, String> collectQuery(List<NameValuePair> args) {
+    val result = new HashMap<String, String>();
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java]: Prefer immutable collections.

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java (hunk 6 above)
Comment (ayang99) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java]: removed in subsequent commit

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java [hunk 8]
```diff
@@ -118,20 +117,13 @@ static boolean isExpired(LocalDateTime expiry) {
...
-    } else {
-      System.out.println("Not expired; still " + ChronoUnit.SECONDS.between(now, expiry) + " seconds to go");
-    }
     return expiry.isBefore(now);
   }
 
   static Map<String, String> collectQuery(List<NameValuePair> args) {
     val result = new HashMap<String, String>();
     for (val pair : args) {
       // store keys as lower case by convention
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java]: Uppercase

File: dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java [hunk 9]
```diff
@@ -118,20 +117,13 @@ static boolean isExpired(LocalDateTime expiry) {
...
-    if (expiry.isBefore(now)) {
-      System.out.println("Expired!");
-    } else {
-      System.out.println("Not expired; still " + ChronoUnit.SECONDS.between(now, expiry) + " seconds to go");
-    }
     return expiry.isBefore(now);
   }
 
   static Map<String, String> collectQuery(List<NameValuePair> args) {
     val result = new HashMap<String, String>();
```
Comment (btiernay) [inline dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/util/PresignedUrlValidator.java]: `ImmutableMap.Builder` or:

```java
import static Collectors.toImmutableMap;
...
args.stream().[stuff].collect(toImmutableMap)
```

**Comment by btiernay** [comment]
Please release and update release notes. Thanks!

**Pull request merged**